[DOCS-1166][DOCS-880][DOCS-1167][DOCS-1168] Update automation docs (#1084)
wandb/docs: content/guides/core/automations/automation-events.md

---
menu:
  default:
    identifier: automation-scopes
    parent: automations
title: Automation events and scopes
weight: 2
---
An automation can run when a specific event occurs at a given scope, either a registry or a project. This page lists the events that can trigger an automation at each scope.

> ## Registry

Reviewer: can you add a gif or screenshot pointing the user to where they can find this scope?

This section describes the scopes and events for an automation in a Registry.

### Scopes
You create a Registry automation in these scopes:

> - A [Registry]({{< relref "/guides/core/registry/">}}) and all of its collections (the default).

Reviewer: ```suggestion
- A [Registry]({{< relref "/guides/core/registry/">}}) and all of its collections (the default). This automation listens for the event taking place on any collection in the given registry. 
```
What does an automation being applied to a registry mean? An automation applied to a registry == an automation applied to all of the collections. Here you say "A Registry and all of its collection" which is a bit confusing. Adding an automation to a registry IS basically bulk adding it to all collections in there (and all future collections added).

Author: But you can scope an automation to only a specific collection. That's what I am trying to say. Let me try again.

Author: PTAL!

- A single collection.

### Events
These events can trigger a Registry automation:
- **Linking a new artifact to a collection**: Test and validate new models or datasets when they are registered.

> - **Adding a new alias to a version of an artifact**: Trigger a special step of your workflow when a new artifact version has a specific alias applied. For example, deploy a model when it has the `deploy` alias applied.

Reviewer: ```suggestion
- **Adding a new alias to a version of an artifact**: Trigger a special step of your workflow when a new artifact version has a specific alias applied. For example, deploy a model when it has the `production` alias applied.
```
aliases typically are a model lifecycle state, and the automation an alias might trigger is the one who's action will move the model into this state.

> ## Project

Reviewer: can you add a gif or screenshot pointing the user to where they can find this scope?

Author: I'm not sure what you mean. Can you say more?

Reviewer: i.e for the scopes section, how one finds project automations. And for registry automations how one finds registry and collection automations.

Author: I didn't add images for these because this is meant to be a reference page. Instead, I gave very consdensed steps for getting to the project or registry, linked out to their main docs pages, and added some links from here to the tasks for creating the automations.

This section describes the scopes and events for an automation in a project.

### Scopes
You can create a project automation in these scopes:
- The project and all of its collections (the default).
- All collections that match the filter you supply.

### Events
These events can trigger a project automation:

> - **Linking a new artifact**: Test and validate new models, datasets or dataset automatically.

Reviewer: ```suggestion
- **Adding a new artifact version to a project collection**: Apply recurring actions to each version of an artifact. For example, start a training job when a new dataset artifact version is created.
```
There is no linking happening in the project. Someone links an artifact in project into the registry, but that is for a registry-scoped automation.

Author: I think I was trying to list out these.events, in the UI: 
![Screenshot 2025-03-26 at 12 40 53 PM](https://github.com/user-attachments/assets/b2297005-8bab-4e1a-aa07-020cff7a190e)
But I think that the UI changed since this copy was written. Let me apply your easy feedback and then I'll look more closely at these.

Author: Done

> - **Creating a new version of an artifact**: Apply recurring actions to each version of an artifact. For example, start a training job when a new dataset artifact version is created.

Reviewer: ```suggestion
```
same as above

Author: Done

> - **Adding a new alias to a version of an artifact**: Trigger a special step of your workflow when a new artifact version in a project or collection has a specific label or alias applied. For example, run a series of downstream processing steps when an artifact has the `test-set-quality-check` alias applied.

Reviewer: ```suggestion
- **Adding a new alias to a version of an artifact**: Trigger a special step of your workflow when a new artifact version in a project or collection has a specific alias applied. For example, run a series of downstream processing steps when an artifact has the `test-set-quality-check` alias applied.
```
new version and alias are 2 separate events. The artifact doesn't need to be new when adding the alias to it / trigger the action. 

We don't have the concept of labels.

Author: Done

---
menu:
  default:
    identifier: automation-scopes
    parent: automations
title: Automation events and scopes
weight: 2
---
{{% pageinfo color="info" %}}
{{< readfile file="/_includes/enterprise-only.md" >}}
{{% /pageinfo %}}

> An automation can run when a specific event occurs at a given scope, either a registry or a project. This page describes the events that can trigger an automation at each scope.

Reviewer 2: Nit: I try to avoid using "run" if it's not in the context of creating a run. (Unfortunate naming of an object).

> An automation can run when a specific event occurs at a given scope, either a registry or a project. This page describes the events that can trigger an automation at each scope.

Reviewer 2: ```suggestion
An automation can start when a specific event occurs within a project's or registrie's  scope. The *scope* of a project refers to [INSERT tech def of scope]. This page describes the events that can trigger an automation within each scope.
```

Author: Done with some tweaking.

Learn more about automations in the [Automations overview]({{< relref "/guides/core/automations/" >}}) or [Create an automation]({{< relref "create-automations/" >}}).

## Registry
This section describes the scopes and events for an automation in a [Registry]({{< relref "/guides/core/registry/">}}).

1. Navigate to the **Registry** App at https://wandb.ai/registry/.
1. Click the name of a registry, then view and create automations in the **Automations** tab.

Learn more about [creating automations]({{< relref "create-automations/" >}}).

### Scopes
You can create a Registry automation at these scopes:

> - A [Registry]({{< relref "/guides/core/registry/">}}). The automation watches for the event taking place on any collection in the registry.
> - A single collection in a registry.

Reviewer 2: A given org can have multiple registries. 

```suggestion
- [Registry]({{< relref "/guides/core/registry/">}}) level: The automation watches for the event taking place on any collection within a specific registry, including collections added in the future.
- Collection level: A single collection in a specific registry.
```

Author: I updated your edit a bit, but I thought that "A Registry" was the same thing as "Registry level: ..... a specific registry." Does that make sense? It seems we are coining "level" when we never had that before.

### Events
A Registry automation can watch for these events:

> - **Linking a new artifact to a collection**: Test and validate new models or datasets when they are registered.

Reviewer 2: Generally a good idea to avoid using the product name as a verb.
```suggestion
- **Linking a new artifact to a collection**: Test and validate new models or datasets when they are added to a registry.
```

> - **Adding a new alias to a version of an artifact**: Trigger a special step of your workflow when a new artifact version has a specific alias applied. For example, deploy a model when it has the `production` alias applied.

Reviewer 2: ```suggestion
- **Adding a new alias to a version of an artifact**: Trigger a specific step of your workflow when a new artifact version has a specific alias applied. For example, deploy a model when it has the `production` alias applied.
```

## Project
This section describes the scopes and events for an automation in a [project]({{< relref "/guides/models/track/project-page.md" >}}).

1. Navigate to your W&B project on the W&B App at `https://wandb.ai/<team>/<project-name>`.
1. View and create automations in the **Automations** tab.

Learn more about [creating automations]({{< relref "create-automations/" >}}).

### Scopes
You can create a project automation at these scopes:

> - The project. The automation watches for the event taking place on any collection in the project.
> - All collections in the project that match the filter you supply.

Reviewer 2: ```suggestion
- Project level: The automation watches for the event taking place on any collection in the project.
- Collection level: All collections in the project that match the filter you specify.
```

Author: Similar feedback as in the other one about adding a new concept "level" (but even more so, since a project can't have multiple projects), but I will take your suggestion.

### Events
A project automation can watch for these events:
- **A new version of an artifact is created in a collection**: Apply recurring actions to each version of an artifact. Specifying a collection is optional. For example, start a training job when a new dataset artifact version is created.

> - **An artifact alias is added**: Trigger a special step of your workflow when a new artifact version in a project or collection has a specific alias applied. For example, run a series of downstream processing steps when an artifact has the `test-set-quality-check` alias applied.

Reviewer 2: Not sure what "special" means? Perhaps "specific" is more appropriate? 

```suggestion
- **An artifact alias is added**: Trigger a specific step of your workflow when a new artifact version in a project or collection has a specific alias applied. For example, run a series of downstream processing steps when an artifact has the `test-set-quality-check` alias applied.
```
